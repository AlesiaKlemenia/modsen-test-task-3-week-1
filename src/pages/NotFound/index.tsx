import { Button } from '@mui/material';

import {
  NotFoundDescription,
  NotFoundTitle,
  NotFoundWrapper,
} from '@/pages/NotFound/styled';

const NotFound = (): JSX.Element => {
  return (
    <NotFoundWrapper>
      <NotFoundTitle variant="h1">404</NotFoundTitle>
      <NotFoundDescription variant="h5" maxWidth="md">
        The page you’re looking for doesn’t exist.
      </NotFoundDescription>
      <Button variant="contained" size="large" href="/">
        Back Home
      </Button>
    </NotFoundWrapper>
  );
};

export default NotFound;
