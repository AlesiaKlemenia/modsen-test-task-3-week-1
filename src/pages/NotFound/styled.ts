import { Box, Typography } from '@mui/material';
import styled from 'styled-components';

export const NotFoundWrapper = styled(Box)`
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  gap: 64px;
  min-height: 100vh;
`;

export const NotFoundTitle = styled(Typography)`
  font-size: 128px;
  text-align: center;
`;

export const NotFoundDescription = styled(Typography)`
  text-align: center;
`;
