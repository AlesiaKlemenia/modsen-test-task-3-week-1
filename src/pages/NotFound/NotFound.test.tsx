import { Button, StyledEngineProvider } from '@mui/material';
import React from 'react';
import renderer from 'react-test-renderer';

import NotFound from '@/pages/NotFound';

describe('NotFound component', () => {
  it('renders correctly', () => {
    const component = renderer.create(
      <StyledEngineProvider injectFirst>
        <NotFound />
      </StyledEngineProvider>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the correct title', () => {
    const component = renderer.create(
      <StyledEngineProvider injectFirst>
        <NotFound />
      </StyledEngineProvider>,
    );
    const instance = component.root;

    const title = instance.findByProps({ variant: 'h1' });
    expect(title.props.children).toBe('404');
  });

  it('renders the correct description', () => {
    const component = renderer.create(
      <StyledEngineProvider injectFirst>
        <NotFound />
      </StyledEngineProvider>,
    );
    const instance = component.root;

    const description = instance.findByProps({ variant: 'h5', maxWidth: 'md' });
    expect(description.props.children).toBe(
      'The page you’re looking for doesn’t exist.',
    );
  });

  it('renders the back home button with correct props', () => {
    const component = renderer.create(
      <StyledEngineProvider injectFirst>
        <NotFound />
      </StyledEngineProvider>,
    );
    const instance = component.root;

    const backButton = instance.findByType(Button);
    expect(backButton.props.variant).toBe('contained');
    expect(backButton.props.size).toBe('large');
    expect(backButton.props.href).toBe('/');
    expect(backButton.props.children).toBe('Back Home');
  });
});
