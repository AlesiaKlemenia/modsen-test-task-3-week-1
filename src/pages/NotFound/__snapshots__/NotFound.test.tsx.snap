// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound component renders correctly 1`] = `
<div
  className="sc-beySPh duKABF MuiBox-root css-0"
>
  <h1
    className="MuiTypography-root MuiTypography-h1 sc-guDLey jSayDs css-o2w69a-MuiTypography-root"
  >
    404
  </h1>
  <h5
    className="MuiTypography-root MuiTypography-h5 sc-dmyCSP cVwgwO css-b4p6uz-MuiTypography-root"
  >
    The page you’re looking for doesn’t exist.
  </h5>
  <a
    className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge css-jh47zj-MuiButtonBase-root-MuiButton-root"
    href="/"
    onBlur={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
  >
    Back Home
  </a>
</div>
`;
